cmd.c

```diff
@@ -743,43 +743,52 @@ static int cmd_loadApp(phfs_conf_t *phfs, const char *name, const char *imap, co
 
 void cmd_app(char *s)
 {
-	u16 cmdArgsc = 0;
-	int i = 0, argID = 0;
+	u16 i, cmdArgsc = 0, argID = 0;
 	char cmdArgs[MAX_CMD_ARGS_NB][LINESZ + 1];
 
 	char *name;
-	unsigned int pos = 0;
+	unsigned int pos = 0, flags = 0;
 	char cmap[8], dmap[8];
 	char appData[3][LINESZ + 1];
 
-	phfs_conf_t phfs;
-
-	phfs.datasz = 0;
-	phfs.dataOffs = 0;
+	phfs_conf_t phfs = { 0 };
 
 	/* Parse command arguments */
-	if (cmd_parseArgs(s, cmdArgs, &cmdArgsc) < 0) {
-		plostd_printf(ATTR_ERROR, "\nWrong arguments!!\n");
+	if (cmd_parseArgs(s, cmdArgs, &cmdArgsc) < 0 && (cmdArgsc < 3 || cmdArgsc > 6))
+		goto errWrongArgs;
+
+	if (cmd_checkDev(cmdArgs[argID++], &phfs.dn) < 0)
 		return;
+
+	/* Check optional flags */
+	if (cmdArgs[argID][0] == '-') {
+		if (!plostd_strcmp(cmdArgs[argID], "-x")) {
```

Comment:
```suggestion
		if ((cmdArgs[argID][1] == 'x') && (cmdArgs[argID][2] == '\0')) {
```
Probably it will be faster.

Reply:
I accepted your sugestion, but modified a little to be case insensitive (valid for ASCII only, use of other char-sets is not considered for plo).